app/src/main/java/de/dhbw/corona_world_app/api/APIManager.java

```diff
@@ -9,142 +9,201 @@
 
         int cnt = 0;
         for (Country country:returnList) {
-            if(popMap.containsKey(country.getName())) {
-                country.setPopulation(popMap.get(country.getName()));
-            } else if(heroToGoogleMap.containsKey(country.getName())) {
-                country.setPopulation(popMap.get(heroToGoogleMap.get(country.getName())));
-            } else {
-                cnt+=1;
-                //System.out.println("\""+country.getName()+"\"s popCount could not be set");
-                Logger.logD("getDataWorld","country \""+country.getName()+"\" has no popCount\nINFO: Try adding an according entry into the <Name,Alias> Map");
+            if(!mapper.isInBlacklist(country.getName())) {
+                if (popMap.containsKey(ISOCountry.valueOf(country.getName()))) {
+                    country.setPopulation(popMap.get(ISOCountry.valueOf(country.getName())));
+                } else {
+                    cnt += 1;
+                    Logger.logD("getDataWorld", "country \"" + country.getName() + "\" has no popCount\nINFO: Try adding an entry into the according Map");
+                }
             }
         }
-        //System.out.println(popMap);
-        //System.out.println("count of countries with no popCount: "+cnt);
         Logger.logD("getDataWorld","count of countries with no popCount: "+cnt);
         return returnList;
     }
 
+    //todo performance
     public List<Country> getData(List<ISOCountry> countryList, List<Criteria> criteriaList, LocalDateTime[] timeFrame){
         Logger.logD("getData","Getting data according to following parameters: "+countryList+" ; "+criteriaList+" ; "+timeFrame);
 
         List<Country> returnList = new ArrayList<>();
-
-        //building url
-        for (ISOCountry isoCountry:countryList) {
-            String url = heroURL;
-            url += "/countries";
-
-            //in/decrease countryList.size if necessary -> todo performance
-            if (countryList != null) {
-                String attachString = "";
-                if(heroToPopMap.containsKey(isoCountry.toString())) {
-                    attachString = heroToPopMap.get(isoCountry.toString());
-                } else {
-                    attachString = isoCountry.toString();
+        if(countryList.size() <= MAX_COUNTRY_LIST_SIZE) {
+            for (ISOCountry isoCountry : countryList) {
+
+                //make api-call
+                Future future = service.submit(new Callable<String>() {
+                                                   @Override
+                                                   public String call() throws Exception {
+                                                       String url = APIs.HEROKU.getUrl();
+                                                       url += APIs.HEROKU.getGetOne();
+
+                                                       if (countryList != null) {
+                                                           String attachString = "";
```

Comment:
Bitte Stringbuilder nutzen

Reply:
Stringbuilder bringt in diesem Zusammenhang nichts siehe https://stackoverflow.com/questions/1532461/stringbuilder-vs-string-concatenation-in-tostring-in-java